DAL/Repositories/InstituteRepository.cs

```diff
@@ -24,109 +23,44 @@ public InstituteRepository(MyContext context)
         public override int Add(Institute institute)
         {
 
-            try
-            {
-                context.Institute.Add(institute);
-                context.SaveChanges();
-            }
-            catch (SqlException sqlEx)
-            {
-                Console.WriteLine(sqlEx.Message);
-            }
-            catch (Exception ex)
-            {
-                Console.WriteLine(ex.Message);
-            }
-            finally
-            {
-
-                Console.WriteLine();
-            }
+            context.Institute.Add(institute);
```

Comment:
Can you call new stored procedure "Institute_LastAdded" here instead of calling context Add ?